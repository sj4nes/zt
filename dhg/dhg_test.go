package dhg

import "testing"

func TestZeroGraph(t *testing.T) {
	g := &Graph{}
	if g.Vertices != nil {
		t.Errorf("expected nil vertices, got %v", g.Vertices)
	}
	if g.Edges != nil {
		t.Errorf("expected nil edges, got %v", g.Edges)
	}
}

func TestGraphToYAML(t *testing.T) {
	g := NewGraph()
	y, err := g.ToYAML()
	if err != nil {
		t.Errorf("unexpected error: %v", err)
	}
	if string(y) != "id: \"16_1\"\nvalue: null\nvertices: null\nedges: null\n" {
		t.Errorf("unexpected yaml: %v", string(y))
	}
}

func TestGraphFmt(t *testing.T) {
	g := NewGraph()
	if g.Fmt() != "î™¢ TODO" {
		t.Errorf("unexpected fmt: %v", g.Fmt())
	}
}

func TestGraphAddVertex(t *testing.T) {
	g := NewGraph()
	v := NewVertex()
	g.AddVertex(v)
	if g.Vertices != nil && len(*g.Vertices) != 1 {
		t.Errorf("expected 1 vertex, got %v", len(*g.Vertices))
	}
	if g.Vertices != nil && (*g.Vertices)[0] != *v {
		t.Errorf("unexpected vertex: %v", (*g.Vertices)[0])
	}
}
