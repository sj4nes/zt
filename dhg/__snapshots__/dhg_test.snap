
[TestGraphToYAML - 1]
id: "16_1"
vertices:
- id: "16_2"
- id: "16_3"
- id: "16_4"
edges:
- id: "16_5"
- id: "16_6"
- id: "16_7"

---

[TestGraphSaveYAML - 1]
id: 16_b
vertices:
- id: 16_c
- id: 16_d
- id: 16_e
edges:
- id: 16_f
- id: 16_g
- id: 16_h

---
